src/features/todo/components/Add/AddTodo.tsx

```diff
@@ -1,13 +1,17 @@
+import { useState } from 'react';
+import { useMutation } from 'react-query';
 import TodoComponent from '@/components/Todo/TodoComponent';
 import { addTodoEndpoint } from '../../../../api';
 
 const AddTodo = () => {
+  const [clearTrigger, setClearTrigger] = useState(false);
+  const addTodoMutation = useMutation(addTodoEndpoint);
```

Comment:
```suggestion
  const addTodoMutation = useMutation(addTodoEndpoint, {
    onSuccess: () => {
      setClearTrigger(true);
    },
  });
```

I would've set the clear trigger here so that it will only reset if the request is successful.